### PR TITLE
Apply visibility: hidden to carousel spacers in amp-carousel 0.2

### PR DESCRIPTION
**Pull request by caroqliu**
Spacer elements are used to establish scroll width / scroll momentum but should not themselves be seen. This change should also save some work in the rendering engine without impacting end user experience. 

Impacts `amp-carousel-0.2` and `amp-base-carousel-0.1`.

Notes:
- Preferred to `content-visibility: hidden` as we will never switch between `hidden` and `visible` states
- Preferred to `display: none` as these elements should still have size (hence the name "spacer"!)

### CODE REVIEW

**Comment by micajuine-ho** [comment]
Thanks for this!

> Preferred to content-visibility: hidden as we will never switch between hidden and visible states
> Preferred to display: none as these elements should still have size (hence the name "spacer"!)

I only see  the addition of `visibility: hidden;`, am I misunderstanding the description?

**Comment by caroqliu** [comment]
> Thanks for this!
> 
> > Preferred to content-visibility: hidden as we will never switch between hidden and visible states
> > Preferred to display: none as these elements should still have size (hence the name "spacer"!)
> 
> I only see the addition of `visibility: hidden;`, am I misunderstanding the description?

Nope. The description is saying that `visibility: hidden` is applied _instead of_ `content-visibility: hidden` or `display: none` for the respective reasons.

File: extensions/amp-base-carousel/0.1/carousel.css [hunk 1]
```diff
@@ -57,17 +57,17 @@
  * calculating the 100% height (or width) of the slide.
 */
 /* Firefox */
-.i-amphtml-carousel-scroll[hide-scrollbar="true"] {
-  scrollbar-width: none; 
+.i-amphtml-carousel-scroll[hide-scrollbar='true'] {
```
Comment (kristoferbaxter) [inline extensions/amp-base-carousel/0.1/carousel.css:60]: This change appears unrelated, was this from prettier formatting?

File: extensions/amp-base-carousel/0.1/carousel.css [hunk 2]
```diff
@@ -85,16 +85,17 @@
...
 }
 
-.i-amphtml-carousel-scroll[user-scrollable="false"] {
+.i-amphtml-carousel-scroll[user-scrollable='false'] {
   overflow: hidden;
 }
 
 .i-amphtml-carousel-spacer {
-  z-index: -1;  
+  visibility: hidden;
```
Comment (kristoferbaxter) [inline extensions/amp-base-carousel/0.1/carousel.css:97]: Does this carry an accessibility concern? I can't imagine it does, but just want to verify.

**Review by kristoferbaxter — APPROVED** [review]
LGTM, two small questions.

File: extensions/amp-base-carousel/0.1/carousel.css (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/0.1/carousel.css:60]: Yea this whole file was formatted in [11b3b3a](https://github.com/ampproject/amphtml/pull/33279/commits/11b3b3a3dafe243baed98d240b1514813fde64e6)

File: extensions/amp-base-carousel/0.1/carousel.css (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-base-carousel/0.1/carousel.css:97]: Confirming there is no impact on accessibility with this change.
